elastic/provider/client.py

```diff
@@ -0,0 +1,75 @@
+from elasticsearch import Elasticsearch
+from flask import current_app as app
+
+client = None
+
+
+class ElasticsearchClient:
+    def __init__(self, connection_params=None, index=None, search_limit=10):
+        if not connection_params:
+            raise ValueError(
+                "No connection parameters provided to the Elasticsearch "
+                "client during initialization."
+            )
+        if not index:
+            raise ValueError(
+                "No index provided to the Elasticsearch "
+                "client during initialization."
+            )
+
+        self.client = Elasticsearch(**connection_params)
+        self.index = index
+        self.search_limit = search_limit
+
+    def search(self, query):
+        es_query_body = {
+            "query": {"multi_match": {"query": query}},
+            "highlight": {
+                "pre_tags": [""],
+                "post_tags": [""],
+                "fields": {"content": {}},
+            },
+        }
+
+        response = self.client.search(
+            index=self.index, body=es_query_body, size=self.search_limit
+        )
+
+        return response["hits"]["hits"]
```

Comment:
should we check that these keys exist? i.e. not None?

Reply:
Done